Spec/Tests/RealtimeClientChannelTests.swift

```diff
@@ -3889,7 +3889,67 @@ class RealtimeClientChannelTests: XCTestCase {
             channel.attach()
         }
     }
+    
+    // RTL13b, RTB1
+    func test__131b__Channel__if_the_channel_receives_a_server_initiated_DETACHED_message_and_if_the_attempt_to_reattach_fails_then_the_channel_will_transition_to_SUSPENDED_state_with_periodic_reattach_with_channelRetryTimeout() {
+        
+        let options = AblyTests.commonAppSetup()
+        options.channelRetryTimeout = 1.0
+        options.autoConnect = false
+        let client = ARTRealtime(options: options)
+        client.internal.setTransport(TestProxyTransport.self)
+        
+        let previousRealtimeRequestTimeout = ARTDefault.realtimeRequestTimeout()
+        ARTDefault.setRealtimeRequestTimeout(1.0)
+        client.connect()
+        
+        defer { ARTDefault.setRealtimeRequestTimeout(previousRealtimeRequestTimeout); client.dispose(); client.close() }
+        
+        let transport = client.internal.transport as! TestProxyTransport
+        transport.actionsIgnored += [.attached]
 
+        expect(client.connection.state).toEventually(equal(ARTRealtimeConnectionState.connected), timeout: testTimeout)
+
+        let channel = client.channels.get(uniqueChannelName())
+        channel.attach()
+        
+        expect(channel.state).to(equal(ARTRealtimeChannelState.attaching))
+
+        let detachedMessageWithError = AblyTests.newErrorProtocolMessage()
+        detachedMessageWithError.action = .detached
+        detachedMessageWithError.channel = channel.name
+        
+        ARTUtils.seedRandom(withNumber: 5)
+        
+        var retryNumber = 1
+        let maxRetryCount = 5
+        
+        let predefinedDelays = [ 0.85, 1.16, 1.633333333333333, 1.712, 1.696 ]
+        var actualDelays = [TimeInterval]()
+        
+        waitUntil(timeout: testTimeout) { done in
+            var suspendedAt: Date!
+            channel.on(.attaching) { stateChange in
+                guard stateChange.previous == .suspended, let suspendedAt = suspendedAt else { return }
+                let actualDelay = Date().timeIntervalSince(suspendedAt)
+                actualDelays.append(actualDelay)
+                retryNumber += 1
+            }
+            channel.on(.suspended) { stateChange in
+                suspendedAt = Date()
+                expect(stateChange.reason).toNot(beNil())
+                if retryNumber > maxRetryCount {
+                    done()
+                }
+            }
+            channel.on(.attached) { _ in
+                fail("Should not receive attached message")
+            }
+            client.internal.transport?.receive(detachedMessageWithError) // force to .suspended
+        }
+        expect(actualDelays).to(beCloseTo(predefinedDelays, within: 0.15))
```

Comment:
The meaning of 0.15 isn’t clear to me – is there a way to make this clearer?

This applies to the similar code in `Spec/Tests/RealtimeClientConnectionTests.swift` too.

Reply:
That was the best accuracy I could get (never hits exact values). With `0.1` fails often.

Reply:
OK, I'll look at the test again once I've got a better idea of [what `predefinedDelays` is](https://github.com/ably/ably-cocoa/pull/1520#discussion_r1036421743).